lib/repl.js

```diff
@@ -689,8 +690,71 @@ function intFilter(item) {
   return /^[A-Za-z_$]/.test(item);
 }
 
+const defaultProperties = [
+  [ Array, Object.getOwnPropertyNames([]).filter(intFilter) ],
+  [ Buffer, Object.getOwnPropertyNames(Buffer.alloc(0)).filter(intFilter) ],
+
+  [ Uint8Array,
+    Object.getOwnPropertyNames(new Uint8Array()).filter(intFilter) ],
+  [ Uint16Array,
+    Object.getOwnPropertyNames(new Uint16Array()).filter(intFilter) ],
+  [ Uint32Array,
+    Object.getOwnPropertyNames(new Uint32Array()).filter(intFilter) ],
+  [
+    Uint8ClampedArray,
+    Object.getOwnPropertyNames(new Uint8ClampedArray()).filter(intFilter) ],
+  [ Int8Array,
+    Object.getOwnPropertyNames(new Int8Array()).filter(intFilter) ],
+  [ Int16Array,
+    Object.getOwnPropertyNames(new Int16Array()).filter(intFilter) ],
+  [ Int32Array,
+    Object.getOwnPropertyNames(new Int32Array()).filter(intFilter) ],
+  [ Float32Array,
+    Object.getOwnPropertyNames(new Float32Array()).filter(intFilter) ],
+  [ Float64Array,
+    Object.getOwnPropertyNames(new Float64Array()).filter(intFilter) ]
```

Comment:
Of all the types enumerated, only plain arrays have own properties by default (`'length'`). No TypedArrays have own properties.